src/esp/sensor/CameraSensor.cpp

```diff
@@ -115,20 +123,29 @@ bool CameraSensor::drawObservation(sim::Simulator& sim) {
   renderTarget().renderEnter();
 
   gfx::RenderCamera::Flags flags;
-  if (sim.isFrustumCullingEnabled())
+  if (sim.isFrustumCullingEnabled()) {
     flags |= gfx::RenderCamera::Flag::FrustumCulling;
+  }
 
-  gfx::Renderer::ptr renderer = sim.getRenderer();
   if (cameraSensorSpec_->sensorType == SensorType::Semantic) {
     // TODO: check sim has semantic scene graph
-    renderer->draw(*this, sim.getActiveSemanticSceneGraph(), flags);
-    if (&sim.getActiveSemanticSceneGraph() != &sim.getActiveSceneGraph()) {
+    bool twoSceneGraphs =
+        (&sim.getActiveSemanticSceneGraph() != &sim.getActiveSceneGraph());
+
+    if (twoSceneGraphs) {
+      VisualSensor::MoveSemanticSensorNodeHelper helper(*this, sim);
+      draw(sim.getActiveSemanticSceneGraph(), flags);
+    } else {
+      draw(sim.getActiveSemanticSceneGraph(), flags);
+    }
+
+    if (twoSceneGraphs) {
       flags |= gfx::RenderCamera::Flag::ObjectsOnly;
-      renderer->draw(*this, sim.getActiveSceneGraph(), flags);
+      draw(sim.getActiveSceneGraph(), flags);
     }
   } else {
-    // SensorType is Depth or any other type
-    renderer->draw(*this, sim.getActiveSceneGraph(), flags);
+    // SensorType is Color, Depth or any other type
+    draw(sim.getActiveSceneGraph(), flags);
```

Comment:
@erikwijmans :
If there is only one scene graph, we assume it is the regular scene graph.
Is there a chance that we will only have semantic scene graph?
also cc: @aclegg3

Reply:
If we only have a one scene graph we'll use it for both RGB and SEMANTIC and all object will be on the correct scene graph for both renders.